RWA arraylias conversion

DanPuzzuoli opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

Converting `rotating_wave_approximation.py` to use arraylias.

I ended up adding the helper functions `_to_dense` and `_to_dense_list` in `arraylias/alias.py` as the `to_array` method had been being used in to convert everything to dense, and we need something analogous. 


### Details and comments

tests that need to be run before merge:

test.dynamics.models (whole folder)
test.dynamics.arraylias (whole folder)
test.dynamics.signals (whole folder)
test.dynamics.pulse (whole folder)

to24toro reviewed (changes requested):
Thanks. Almost looking good to me.

- Regarding alias.py, it might be better to add tests about `_to_dense`and `_to_dense_list`.

DanPuzzuoli commented on qiskit_dynamics/models/rotating_wave_approximation.py:
```diff
@@ -235,10 +244,10 @@ def jax_transformable_func(t):
-    frame_freqs: Array,
+    frame_freqs: ArrayLike,
     cutoff_freq: float,
 ):
```
The return type for `get_rwa_operators` should be `ArrayLike` and the return type for `get_rwa_signals` is `SignalList`. Just added these type hints.

to24toro replied:
Thank you. Regarding `frame_freqs`, it's my mistake and I referred a wrong part.

DanPuzzuoli commented:
@to24toro I've updated the type hints and added tests for `_to_dense` and `_to_dense_list`.

I had to force push as I'm working from my personal laptop and had forgotten to update the `git config` with my username on commit [7d21240](https://github.com/Qiskit-Extensions/qiskit-dynamics/pull/296/commits/7d21240b02152c9e94cde1868678e8141ca0c3ca), which triggered the CLA message.

to24toro reviewed (approved):
LGTM!
